src/app/catalog/catalog.component.ts

```diff
@@ -1,14 +1,49 @@
 import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+
+import { Book } from './catalog';
+import { CatalogService } from './catalog.service';
 
 @Component({
   selector: 'app-catalog',
   templateUrl: './catalog.component.html',
   styleUrls: ['./catalog.component.scss'],
 })
 export class CatalogComponent implements OnInit {
-  bookId: number = 2;
+  books: Book[] = [];
+  filteredBooks: Book[] = [];
+
+  myControl = new FormControl();
+  options: string[] = ['One', 'Two', 'Three'];
+  filteredOptions: Observable<string[]> | undefined;
 
-  constructor() {}
+  constructor(private catalogService: CatalogService) {}
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.getBooks();
+  }
+  getBooks(): void {
+    this.catalogService.getCatalog().subscribe((books: Book[]) => {
+      this.books = books;
+      this.options = books.map((book) => book.title.toString());
+      this.filteredOptions = this.myControl.valueChanges.pipe(
+        startWith(''),
+        map((value) => {
+          this.filteredBooks = this.books.filter((book) =>
+            book.title.toLowerCase().includes(value.toLowerCase())
+          );
+          return this._filter(value);
+        })
+      );
+    });
+  }
+  private _filter(value: string): string[] {
```

Comment:
filtering the books should be implemented via PipeTransform (custom FilterBook pipe)